### PR TITLE
feat(vue-demo-store): add select for account type & company fields

### LINKED ISSUE #780: [FEATURE] add choice of account type in account registration
### Description

enable to choose a type of an account: private or commercial. show additional fields if needed (company name, VAT id...).

image

### Use Case

 use default storefront as a benchmark 

### Proposed Solution

_No response_

### Alternatives Considered

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by florianliebig**
### Description

Closes #780 

### Type of change

New feature (non-breaking change which adds functionality)

### ToDo's

- [x] Changeset file provided [read more](https://github.com/shopware/frontends/blob/main/CONTRIBUTION.md#changelog-preparation)

- [x] E2E-Tests added/updated

### Screenshots (if applicable)

2024-03-23 at 13 58

2024-03-23 at 13 59

### CODE REVIEW

File: templates/vue-demo-store/i18n/pl-PL/form.json [hunk 1]
```diff
@@ -8,6 +13,8 @@
     "lastNamePlaceholder": "Wpisz nazwisko...",
     "country": "Kraj",
     "chooseCountry": "Wybierz kraj...",
+    "company": "Firmę",
```
Comment (mdanilowicz) [inline templates/vue-demo-store/i18n/pl-PL/form.json:16]:
```suggestion
    "company": "Firma",
```

File: templates/vue-demo-store/i18n/pl-PL/form.json [hunk 2]
```diff
@@ -8,6 +13,8 @@
...
+    "companyPlaceholder": "Wprowadź firmę...",
```
Comment (mdanilowicz) [inline templates/vue-demo-store/i18n/pl-PL/form.json:17]:
```suggestion
    "companyPlaceholder": "Wprowadź nazwę firmy...",
```

File: templates/vue-demo-store/i18n/pl-PL/form.json [hunk 3]
```diff
@@ -1,5 +1,10 @@
 {
   "form": {
+    "accountType": {
+      "title": "Typ konta",
+      "private": "Osobą prywatną",
+      "business": "Klientem biznesowym"
```
Comment (mdanilowicz) [inline templates/vue-demo-store/i18n/pl-PL/form.json:6]:
```suggestion
      "business": "Klient biznesowy"
```

File: templates/vue-demo-store/i18n/pl-PL/form.json [hunk 4]
```diff
@@ -1,5 +1,10 @@
 {
   "form": {
+    "accountType": {
+      "title": "Typ konta",
+      "private": "Osobą prywatną",
```
Comment (mdanilowicz) [inline templates/vue-demo-store/i18n/pl-PL/form.json:5]:
```suggestion
      "private": "Osoba prywatna",
```

**Review by mdanilowicz — APPROVED** [review]
Nice! Please just accept language changes and merge the main branch. Thank you!

**Review by patzick — APPROVED** [review]

**Pull request merged**